coding-standard/action.yml

```diff
@@ -0,0 +1,138 @@
+    default: "5"
+    description: "The minimum severity required to display an error"
+
+  baseline_version:
+    type: string
+    required: false
+    default: "1.1.2"
+    description: "The version of the phpcs baseline to use. if not required it uses default one"
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout Project
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        path: standard
+
+    - name: Set PHP Version
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ inputs.php_version }}
+        tools: composer:v${{ inputs.composer_version }}
+        coverage: none
+
+    - name: Install Coding Standard && Codesniffer baseline
+      shell: bash
+      working-directory: standard
+      run: |
+        git config --global advice.detachedHead false
+        composer require "magento/magento-coding-standard:${{ inputs.version || '*' }}"
+        composer config --no-plugins allow-plugins.digitalrevolution/php-codesniffer-baseline true
+        composer require --dev "digitalrevolution/php-codesniffer-baseline:${{ inputs.baseline_version }}"
+
+    - name: Register Coding Standard
+      shell: bash
+      working-directory: standard
+      run: vendor/bin/phpcs --config-set installed_paths ${{ github.workspace }}/standard/vendor/magento/magento-coding-standard,${{ github.workspace }}/standard/vendor/phpcompatibility/php-compatibility
+
+    - name: Get all changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+
+    - name: Checkout - Before Merge
+      shell: bash
+      working-directory: standard
+      id: previous-commit-files
+      run: |
+        if ${{ github.event_name == 'pull_request' }}; then
+          git checkout ${{ github.event.pull_request.base.ref }}
+        else
+          git checkout ${{ github.event.before }}
+        fi
+
+        for file in ${{ steps.changed-files.outputs.all_modified_files }}; do
+          echo $file >> ${{ github.workspace }}/file_lists.txt
+        done
+        cat ${{ github.workspace }}/file_lists.txt
+
+    - name: Filter php files and execute phpcs - Before Merge
+      shell: bash
+      id: phpcs_before
+      working-directory: standard
+      run: |
+        grep -i "\.php$" ${{ github.workspace }}/file_lists.txt > ${{ github.workspace }}/phpcs_files.txt
+        grep -i "\.phtml$" ${{ github.workspace }}/file_lists.txt >> ${{ github.workspace }}/phpcs_files.txt
+
+        echo $(php vendor/bin/phpcs --standard=Magento2 \
+        $([ -n "${{ inputs.severity }}" ] && echo "--severity=${{ inputs.severity }}") \
+        $([ -n "${{ inputs.warning_severity }}" ] && echo "--warning-severity=${{ inputs.warning_severity }}") \
+        $([ -n "${{ inputs.error_severity }}" ] && echo "--error-severity=${{ inputs.error_severity }}") \
+        --report=\\DR\\CodeSnifferBaseline\\Reports\\Baseline --file-list=${{ github.workspace }}/phpcs_files.txt) > ${{ github.workspace }}/phpcs_old.xml
```

Comment:
The file needs to be called `phpcs.baseline.xml`, so that it is automatically picked up: https://github.com/123inkt/php-codesniffer-baseline#usage

Reply:
okay will check that

Reply:
Ideally it should work as written but when we execute that file not generated automatically. when we use that report-file=phpcs.baseline.xml - the command gives an error.

Reply:
Which error? Maybe you need something like `report-file=${{ github.workspace }}/phpcs.baseline.xml`?